### PR TITLE
Bug 1322759 - Add description field to spark jobs

### PR DESCRIPTION
**Pull request by robhudson**
The new field as it appears on the form...

screenshot 2017-01-13 18 12 11

### CODE REVIEW

File: atmo/forms/mixins.py [hunk 1]
```diff
@@ -1,10 +1,10 @@
 # This Source Code Form is subject to the terms of the Mozilla Public
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, you can obtain one at http://mozilla.org/MPL/2.0/.
-from collections import OrderedDict
 import uuid
+from collections import OrderedDict
+
 from django import forms
-from django.core.exceptions import ImproperlyConfigured
```
Comment (jezdez) [inline atmo/forms/mixins.py]: ❤️

File: atmo/jobs/forms.py [hunk 2]
```diff
@@ -26,9 +26,20 @@ class BaseSparkJobForm(FormControlFormMixin, CachedFileModelFormMixin,
             'class': 'identifier-taken-check',
             'data-identifier-taken-check-url': reverse_lazy('jobs-identifier-taken'),
         }),
-        help_text='A brief description of the scheduled Spark job\'s purpose, '
+        help_text='A slug-like string to identify your Spark job, '
```
Comment (jezdez) [inline atmo/jobs/forms.py]: "slug-like" is a bit technical and not likely to be understood by the ATMO users, can we find a middleground here?

File: atmo/jobs/models.py [hunk 3]
```diff
@@ -38,6 +38,10 @@ class SparkJob(EMRReleaseModel, CreatedByModel):
         help_text="Job name, used to uniqely identify individual jobs.",
         unique=True,
     )
+    description = models.TextField(
+        help_text='Job description.',
+        default='',
```
Comment (jezdez) [inline atmo/jobs/models.py]: Is the empty default needed for this?

File: atmo/jobs/forms.py (hunk 2 above)
Comment (robhudson) [inline atmo/jobs/forms.py]: Question: Does this string need to be a slug? Or can it just be a short string with spaces or capitals? I didn't see any slug processing but all the examples in tests are "slug-like".

File: atmo/jobs/models.py (hunk 3 above)
Comment (robhudson) [inline atmo/jobs/models.py]: That default is needed for the migration for existing records in the database. Is there a better place for that to live?

File: atmo/jobs/forms.py (hunk 2 above)
Comment (jezdez) [inline atmo/jobs/forms.py]: Yeah, the [job form will validate](https://github.com/mozilla/telemetry-analysis-service/blob/7ce65254adf96217931ca3f90d0b6ef9970b9e73/atmo/jobs/forms.py#L23) that with `[\w-]{1,100}$` and [remove whitespaces and underscores on the client side](https://github.com/mozilla/telemetry-analysis-service/blob/7ce65254adf96217931ca3f90d0b6ef9970b9e73/atmo/static/js/forms.js#L36-L37) when it checks for identifier availability.

So I'd say it *is* slug-like, but think the term is not easy to understand -- hence the more general term "identifier". Maybe we should call it "label-like"?

File: atmo/jobs/models.py (hunk 3 above)
Comment (jezdez) [inline atmo/jobs/models.py]: Ah, nope, totally blanked on that one. Carry one :)

**Comment by robhudson** [comment]
The help text has been updated. Let me know if that works.

**Pull request merged**